### PR TITLE
Special days

### PR DESCRIPTION
**Pull request by ntm5**
Basic implementation of special days for now. Incomplete needs work.

### CODE REVIEW

**Comment by ntm5** [comment]
Completed:
- Basic implementation
- Menu
Todo:
- Actually create the special days

**Comment by ntm5** [comment]
Warday and Freeday should have been implemented. Needs to be overlooked

**Comment by ntm5** [comment]
Should be complete. Awaiting comments.

File: mod/Jailbreak.SpecialDay/SpecialDays/Warday.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+using Jailbreak.Public.Mod.SpecialDays;
+using Microsoft.VisualBasic.CompilerServices;
+using Timer = CounterStrikeSharp.API.Modules.Timers.Timer;
+
+namespace Jailbreak.SpecialDay.SpecialDays;
+
+public class Warday : ISpecialDay
+{
+    public string Name => "Warday";
+    public string Description => "The guards can kill prisoners without any consequences.";
```
Comment (jithatsonei) [inline mod/Jailbreak.SpecialDay/SpecialDays/Warday.cs:16]: Improve these descriptions

File: mod/Jailbreak.SpecialDay/SpecialDays/Warday.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    private void AddTimers()
+    {
+        timer1 = _plugin.AddTimer(1f, () =>
+        {
+            foreach (var player in Utilities.GetPlayers()
+                         .Where(player => player.IsReal()))
+            {
+                if (timer == 3 || player.Team == CsTeam.CounterTerrorist) player.UnFreeze();
```
Comment (jithatsonei) [inline mod/Jailbreak.SpecialDay/SpecialDays/Warday.cs:50]: think about adding a check to see if they're already somehow unfrozen here

File: mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using Jailbreak.Public.Extensions;
+using Jailbreak.Public.Mod.SpecialDays;
+using Timer = CounterStrikeSharp.API.Modules.Timers.Timer;
+
+namespace Jailbreak.SpecialDay.SpecialDays;
+
+public class FreeForAllDay : ISpecialDay
+{
+    public string Name => "Warday";
+    public string Description => "The guards can kill prisoners without any consequences.";
```
Comment (jithatsonei) [inline mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs:15]: Improve description

File: mod/Jailbreak.SpecialDay/SpecialDayHandler.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        return _isSpecialDayActive;
+    }
+
+    public bool StartSpecialDay(string name)
+    {
+        if (_isSpecialDayActive || !CanStartSpecialDay()) return false;
+        
+        var fullName = "Jailbreak.SpecialDay.SpecialDays";
+        var q = from t in Assembly.GetExecutingAssembly().GetTypes()
+            where t.IsClass && t.Namespace == fullName && t.GetInterface("ISpecialDay") != null
```
Comment (jithatsonei) [inline mod/Jailbreak.SpecialDay/SpecialDayHandler.cs:64]: this down to line 72 - improve variable names

File: public/Jailbreak.Public/Jailbreak.Public.csproj [hunk 5]
```diff
@@ -7,7 +7,7 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="CounterStrikeSharp.API" Version="1.0.202"/>
+        <PackageReference Include="CounterStrikeSharp.API" Version="1.0.215"/>
```
Comment (jithatsonei) [inline public/Jailbreak.Public/Jailbreak.Public.csproj:10]: update to latest please

File: mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        {
+            var max = spawn.Count;
+            
+            var index = new Random().Next(0, max);
+            
+            player.Teleport(spawn[index].AbsOrigin);
+            player.Freeze();
+        }
+        
+        Server.ExecuteCommand("mp_friendlyfire 1");
```
Comment (jithatsonei) [inline mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs:43]: why not use cvar system?

File: mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    private void AddTimers()
+    {
+        timer1 = _plugin.AddTimer(1f, () =>
+        {
+            foreach (var player in Utilities.GetPlayers()
+                         .Where(player => player.IsReal()))
+            {
+                if (timer != 5) return;
+                
+                player.UnFreeze();
```
Comment (jithatsonei) [inline mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs:57]: considering adding already unfrozen check

**Review by jithatsonei — COMMENTED** [review]
thanks for the PR, very good start - just a few minor changes needed id say

File: mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs (hunk 6 above)
Comment (ntm5) [reply mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs:43]: How

Comment (jithatsonei) [reply mod/Jailbreak.SpecialDay/SpecialDays/FreeForAllDay.cs:43]: https://github.com/Quantor97/CS2-RollTheDice-Plugin/blob/61a2be19d6bab18302c775a293ca772fb4d5f538/effects/EffectVampire.cs#L24-L28

**Pull request merged**